### PR TITLE
refactor async cancel

### PR DESCRIPTION
**Pull request by vincerubinetti**
- refactor fetches using the generator pattern described here https://dev.to/chromiumdev/cancellable-async-functions-in-javascript-5gp7
- adds fetch cancellation (in-flight) with AbortController

### CODE REVIEW

File: src/actions/fetch.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import { createAction } from 'redux-actions';
+
+import { sleep } from '../util/debug.js';
+import { isArray } from '../util/types.js';
+import { isObject } from '../util/types.js';
+
+// replacement for redux-thunk-actions
```
Comment (dongbohu) [inline src/actions/fetch.js:7]: Can you also add the article's URL in the comments here? I found it very helpful to understand the code here.

File: src/pages/genes/search-results/multi-row/index.js [hunk 2]
```diff
@@ -88,10 +88,12 @@ let MultiRow = ({ search, selectGene, deselectGene }) => {
     <>
       <div className='gene_search_result_multi'>
         <div
-          className='gene_search_result_multi_query text_small'
+          className='gene_search_result_multi_query text_small '
```
Comment (dongbohu) [inline src/pages/genes/search-results/multi-row/index.js]: Any reason why there is a space character at the end?

**Review by dongbohu — COMMENTED** [review]
Left two comments.

**Comment by vincerubinetti** [comment]
> Left two comments.

Resolved

**Review by dongbohu — APPROVED** [review]

**Pull request merged**